vllm/v1/core/kv_cache_coordinator.py

```diff
@@ -388,7 +388,9 @@ def get_kv_cache_coordinator(
         kv_cache_config: KVCacheConfig, max_model_len: int, use_eagle: bool,
         enable_caching: bool, caching_hash_fn: Callable,
         enable_kv_cache_events: bool) -> KVCacheCoordinator:
-    if not enable_caching:
+    if not enable_caching or len(kv_cache_config.kv_cache_groups) == 0:
+        # We instantiate this coordinator also for attention free models that
```

Comment:
Do we need this given prefix caching is disabled here for models that don’t use last pooling method?
https://github.com/maxdebayser/vllm/blob/221f013922c0c118b682d294755e69990b2c43ed/vllm/config.py#L4505

Reply:
Perhaps the safest thing to do is to disable prefix-caching in `VllmConfig.__posto_init__` right away for any attention free models and then yes, we could just rely on `enable_caching` as you suggest.